src/Csv.php

```diff
@@ -540,6 +540,41 @@ public function auto($file = null, $parse = true, $search_depth = null, $preferr
         return $this->delimiter;
     }
 
+    /**
+     * Get total number of rows in csv without parsing whole data.
+     *
+     * @return bool|int
+     */
+    public function getTotalRowCount(){
+        if (empty($this->file_data)){
+            return false;
+        }
+
+        $this->_detect_and_remove_sep_row_from_data($this->file_data);
+
+        $pattern = sprintf('/("[^%s]*")|[^%s]*/i',$this->enclosure, $this->enclosure);
```

Comment:
1. you can re-use the `sprintf` argument with the `%1$s` syntax. See Example #3 in http://de2.php.net/manual/en/function.sprintf.php
2. shouldn't all quotes `"` be replaced by the actual `$this->enclosure` if people used e.g. a single apostrophe `'` instead?
3. I suspect the `|[^%s]*` part is not needed because you call `continue;` for those
4. Since there is nothing to replace for empty cells like this `""` you could use the `+` instead of `*` to get at least a single character.
3. Please add spaces after commas and before curly brackets. Thanks!

Reply:
All done. Thanks.